### PR TITLE
[RA2] add LB specs

### LINKED ISSUE #3316: [RA2] Add Load Balancer specs
apart from service type load balancer
what are the load balancer features that must be supported? protocols? session persistency? bgp?

### PR DESCRIPTION
**Pull request by rgstori**
fixes #3316 
have added 3 specs for load balancer:

- one for the LB itself and its API
- one for active active via BGP
- one for fast failover / HA - ie to prevent ARP VIPs

may also add:

- session stickiness / persistence
- SCTP
- multihoming / VRF support

### CODE REVIEW

**Review by CsatariGergely — APPROVED** [review]

**Review by petorre — APPROVED** [review]

File: doc/ref_arch/kubernetes/chapters/chapter04.rst [hunk 1]
```diff
@@ -543,6 +543,21 @@ the following specifications:
        :cite:t:`multi-net-spec`.
      - gen.ost.01 in :ref:`chapters/chapter02:kubernetes architecture requirements`
      - :cite:t:`anuket-ri2` Chapter 4, section Installation on Bare Metal Infratructure
+   * - ra2.ntw.017
+     - Kubernetes Load Balancer - API
+     - The networking solution deployed within the implementation must include a L4 (TCP/UDP) Load Balancer to steer inbound traffic across the primary interfaces of multiple CNF pods. The solution must support Service type Loadbalancer API and may support the Gateway API additionally.
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter04.rst:548]: There are multiple requirements here, I recommend we split:
-  The networking solution deployed within the implementation must include an L4 (TCP/UDP) Load Balancer to steer inbound traffic across the primary interfaces of multiple CNF pods.
- The Load Balancer solution deployed per ra2.ntw.017 must support Service type Loadbalancer API 
- The Load Balancer solution deployed per ra2.ntw.017 may support the Gateway API additionally.

File: doc/ref_arch/kubernetes/chapters/chapter04.rst [hunk 2]
```diff
@@ -543,6 +543,21 @@ the following specifications:
...
+     - inf.ntw.15 in :ref:`chapters/chapter02:kubernetes architecture requirements`
+     - 
+   * - ra2.ntw.018
+     - Kubernetes Load Balancer - Advertisements & Active/active Multipath
+     - The networking solution deployed within the implementation must include a L4 (TCP/UDP) Load Balancer capable of advertising the IPs of Services to external networks. To achieve an active/active design, the advertisements must support multi-path, allowing the same service IP to be advertised by multiple cluster nodes (where pods serving traffic reside) as next-hops.
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter04.rst:553]: This requirement is also two requirements where the second is specifying an implementation method. As <ra2.ntw.017> is a must, think we can just refer to it
 - The Load Balancer solution deployed per ra2.ntw.017 must be capable of advertising the IPs of Services to external networks. 
 - In an active/active design, the Load Balancer solution deployed per ra2.ntw.017 must support multi-path advertisements, allowing the same service IP to be advertised by multiple cluster nodes.

File: doc/ref_arch/kubernetes/chapters/chapter04.rst [hunk 3]
```diff
@@ -543,6 +543,21 @@ the following specifications:
...
+     - inf.ntw.15 in :ref:`chapters/chapter02:kubernetes architecture requirements`
+     - 
+   * - ra2.ntw.019
+     - Kubernetes Load Balancer - High Availability
+     - The networking solution deployed within the implementation must include a L4 (TCP/UDP) Load Balancer capable of fast failover. Upon node or pod failure, traffic should be redirected (i.e. advertisements/routes must be updated) in less than 5 seconds to prevent application failure.
```
Comment (pgoyal01) [inline doc/ref_arch/kubernetes/chapters/chapter04.rst:558]: Application failure would depend upon the nature of the application.

```suggestion
     - The Load Balancer solution deployed per ra2.ntw.017 must, upon node or pod failure, redirect traffic (i.e., advertisements/routes must be updated) in less than 5 seconds.
```

**Review by pgoyal01 — CHANGES_REQUESTED** [review]

File: doc/ref_arch/kubernetes/chapters/chapter04.rst (hunk 1 above)
Comment (rgstori) [reply doc/ref_arch/kubernetes/chapters/chapter04.rst:548]: agree to split in 3

File: doc/ref_arch/kubernetes/chapters/chapter04.rst (hunk 2 above)
Comment (rgstori) [reply doc/ref_arch/kubernetes/chapters/chapter04.rst:553]: agree to split

File: doc/ref_arch/kubernetes/chapters/chapter04.rst (hunk 3 above)
Comment (rgstori) [reply doc/ref_arch/kubernetes/chapters/chapter04.rst:558]: what about:
```suggestion
     - The networking solution deployed per `ra2.ntw.017` must be capable of fast failover. Upon node or pod failure, it must redirect traffic (i.e., advertisements/routes must be updated) in less than 5 seconds.
```

File: doc/ref_arch/kubernetes/chapters/chapter04.rst (hunk 1 above)
Comment (CsatariGergely) [inline doc/ref_arch/kubernetes/chapters/chapter04.rst:548]:
```suggestion
     - The networking solution deployed within the implementation must include a L4 (TCP/UDP (excluding QUIC)) Load Balancer to steer inbound traffic across the primary interfaces of multiple CNF pods. The solution must support Service type Loadbalancer API and may support the Gateway API additionally.
```

**Review by CsatariGergely — CHANGES_REQUESTED** [review]

File: doc/ref_arch/kubernetes/chapters/chapter04.rst (hunk 1 above)
Comment (rgstori) [reply doc/ref_arch/kubernetes/chapters/chapter04.rst:548]: QUIC as a simple UDP session should be supported?

Comment (rgstori) [reply doc/ref_arch/kubernetes/chapters/chapter04.rst]: fixed in https://github.com/anuket-project/anuket-specifications/pull/3415/commits/55bed7d86d0e8d7ed2b4985a727d570978957848

File: doc/ref_arch/kubernetes/chapters/chapter04.rst (hunk 2 above)
Comment (rgstori) [reply doc/ref_arch/kubernetes/chapters/chapter04.rst]: fixed in https://github.com/anuket-project/anuket-specifications/pull/3415/commits/55bed7d86d0e8d7ed2b4985a727d570978957848

File: doc/ref_arch/kubernetes/chapters/chapter04.rst (hunk 3 above)
Comment (rgstori) [reply doc/ref_arch/kubernetes/chapters/chapter04.rst]: fixed in https://github.com/anuket-project/anuket-specifications/pull/3415/commits/55bed7d86d0e8d7ed2b4985a727d570978957848

**Review by CsatariGergely — APPROVED** [review]

**Pull request merged**